### PR TITLE
Task/authentication

### PR DESCRIPTION
**Pull request by ViktorKapra**

### CODE REVIEW

File: E-Commerce-Server/E-Commerce-Data/EComUser.cs
Comment (hottab9595) [inline E-Commerce-Server/E-Commerce-Data/EComUser.cs:1]: Please move this file to a new Models folder

File: E-Commerce-Server/E-Commerce-Data/EComRole.cs
Comment (hottab9595) [inline E-Commerce-Server/E-Commerce-Data/EComRole.cs:1]: Please move this file to a new Models folder

File: E-Commerce-Server/E-Commerce-Data/EComUser.cs
Comment (hottab9595) [inline E-Commerce-Server/E-Commerce-Data/EComUser.cs:1]: Please move this file to a new Models folder

File: E-Commerce-Server/E-Commerce-Server-Tests/APITests/ValidationTest.Login.cs [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+using System.ComponentModel;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace ECom.Test
+{
+
```
Comment (hottab9595) [inline E-Commerce-Server/E-Commerce-Server-Tests/APITests/ValidationTest.Login.cs]: Remove extra space

File: E-Commerce-Server/E-Commerce-Server/Controllers/HomeController.cs [hunk 2]
```diff
@@ -1,5 +1,7 @@
 ﻿using AutoMapper;
 using Azure.Core;
+using Microsoft.AspNetCore.Authentication.Cookies;
```
Comment (hottab9595) [inline E-Commerce-Server/E-Commerce-Server/Controllers/HomeController.cs]: Please remove unused links and empty spaces

File: E-Commerce-Server/E-Commerce-Server/DTOs/AuthenticationDTO/EmailConfirmDTO.cs [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+namespace ECom.API.DTOs.AuthenticationDTO
+{
+    public class EmailConfirmDTO
+    {
+        [Required]
+        [StringLength(320, ErrorMessage = "Email address is too long")]
+        [RegularExpression(ValidationConsts.EmailRegex,
+          ErrorMessage = "Email addres is not correct")]
+        public string Email { get; set; }
```
Comment (hottab9595) [inline E-Commerce-Server/E-Commerce-Server/DTOs/AuthenticationDTO/EmailConfirmDTO.cs:12]: In future you can use = default if it is needed.

File: E-Commerce-Server/E-Commerce-Server/Mapper/MappingProfile.cs [hunk 4]
```diff
@@ -1,9 +1,17 @@
...
+using ECom.API.DTOs.AuthenticationDTO;
+using ECom.BLogic.Services.Models;
+using Microsoft.AspNetCore.Identity;
 namespace ECom.API.Mapper
 {
     public class MappingProfile : Profile
     {
-        public MappingProfile() { }
+        public MappingProfile() {
+
```
Comment (hottab9595) [inline E-Commerce-Server/E-Commerce-Server/Mapper/MappingProfile.cs]: remove extra space

File: E-Commerce-Server/E-Commerce-Server/Program.cs [hunk 5]
```diff
@@ -1,5 +1,6 @@
 using AutoMapper;
-using E_Commerce_Data;
+using ECom.Configuration.Seeding;
```
Comment (hottab9595) [inline E-Commerce-Server/E-Commerce-Server/Program.cs:1]: remove unused links and extra spaces

File: E-Commerce-Server/E-Commerce-Server/Startup.cs [hunk 6]
```diff
@@ -35,29 +45,57 @@ public void ConfigureServices(IServiceCollection services)
...
 
             var connectionString = Configuration["ConnectionStrings:DefaultConnection"] ?? throw new InvalidOperationException("Connection string 'DefaultConnection' not found.");
 
-           
+
             services.AddDbContext<ApplicationDbContext>(options =>
                 options.UseSqlServer(connectionString));
 
-            services.AddDatabaseDeveloperPageExceptionFilter();
+            //services.AddDatabaseDeveloperPageExceptionFilter();
```
Comment (hottab9595) [inline E-Commerce-Server/E-Commerce-Server/Startup.cs]: please remove

File: E-Commerce-Server/E-Commerce-Server/Startup.cs [hunk 7]
```diff
@@ -35,29 +45,57 @@ public void ConfigureServices(IServiceCollection services)
               .WriteTo.Logger(l => l
                   .Filter.ByIncludingOnly(x => x.Level == Serilog.Events.LogEventLevel.Fatal)
                   .WriteTo.File(".\\Logs\\log_fatal.txt", rollingInterval: RollingInterval.Day))
+              .WriteTo.TestCorrelator()
               .CreateLogger();
 
+            // services.AddAuthentication().AddIdentityCookies();
```
Comment (hottab9595) [inline E-Commerce-Server/E-Commerce-Server/Startup.cs]: please remove

File: E-Commerce-Server/ECom.BLogic/Services/Authentication/AuthService.cs [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+
+
+        public AuthService(SignInManager<EComUser> signInManager,
+             UserManager<EComUser> userManager, IEmailService emailService)
+        {
+            _signInManager = signInManager;
+            _userManager = userManager;
+            _emailService = emailService;
+        }
+        private string EncodeToken(string token) => WebEncoders.Base64UrlEncode(Encoding.UTF8.GetBytes(token));
```
Comment (hottab9595) [inline E-Commerce-Server/ECom.BLogic/Services/Authentication/AuthService.cs]: I prefer to use extension methods for this

File: E-Commerce-Server/ECom.BLogic/Services/Authentication/AuthService.cs [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+            {   
+                Log.Logger.Information("User logged in.");
+            }
+            return result;
+        }
+        public async Task<IdentityResult> RegisterAsync(UserCredentials credentials)
+        {
+
+            
+           var user = Activator.CreateInstance<EComUser>();
```
Comment (hottab9595) [inline E-Commerce-Server/ECom.BLogic/Services/Authentication/AuthService.cs]: please format this class

File: E-Commerce-Server/ECom.BLogic/Services/Authentication/IAuthService.cs [hunk 10]
```diff
@@ -0,0 +1,20 @@
+﻿using ECom.BLogic.Services.Models;
```
Comment (hottab9595) [inline E-Commerce-Server/ECom.BLogic/Services/Authentication/IAuthService.cs]: Usually we put interfaces in the separate folder. In your case like Services/Interfaces, not near BL class

File: E-Commerce-Server/ECom.BLogic/Services/EmailService/EmailService.cs [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+using Serilog;
+using Serilog.Context;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Net;
+using System.Net.Mail;
+using System.Text;
+using System.Threading.Tasks;
```
Comment (hottab9595) [inline E-Commerce-Server/ECom.BLogic/Services/EmailService/EmailService.cs]: Make formatting please

File: E-Commerce-Server/ECom.BLogic/Services/EmailService/IEmailService.cs [hunk 12]
```diff
@@ -0,0 +1,14 @@
+﻿using Microsoft.AspNetCore.Identity.UI.Services;
```
Comment (hottab9595) [inline E-Commerce-Server/ECom.BLogic/Services/EmailService/IEmailService.cs]: Move to the Interfaces folder

File: E-Commerce-Server/ECom.Constants/ValidationConsts.cs [hunk 13]
```diff
@@ -0,0 +1,8 @@
+﻿namespace ECom.Constants
+{
+    public static class ValidationConsts
+    {
+        public const string EmailRegex = "^[^@\\s]+@[^@\\s]+\\.[^@\\s]+$";
```
Comment (hottab9595) [inline E-Commerce-Server/ECom.Constants/ValidationConsts.cs]: Usually the naming for constants is uppercase, like EMAIL_REGEX, but it is okay if you prefer this option

File: E-Commerce-Server/E-Commerce-Data/EComRole.cs
Comment (ViktorKapra) [reply E-Commerce-Server/E-Commerce-Data/EComRole.cs:1]: Created new Account folder. Class moved to Account folder.

File: E-Commerce-Server/E-Commerce-Data/EComUser.cs
Comment (ViktorKapra) [reply E-Commerce-Server/E-Commerce-Data/EComUser.cs:1]: Class moved to Account folder.

File: E-Commerce-Server/E-Commerce-Server-Tests/APITests/ValidationTest.Login.cs (hunk 1 above)
Comment (ViktorKapra) [reply E-Commerce-Server/E-Commerce-Server-Tests/APITests/ValidationTest.Login.cs]: Extra space removed.

File: E-Commerce-Server/E-Commerce-Server/Controllers/HomeController.cs (hunk 2 above)
Comment (ViktorKapra) [reply E-Commerce-Server/E-Commerce-Server/Controllers/HomeController.cs]: Unused links removed.

File: E-Commerce-Server/ECom.BLogic/Services/Authentication/IAuthService.cs (hunk 10 above)
Comment (ViktorKapra) [reply E-Commerce-Server/ECom.BLogic/Services/Authentication/IAuthService.cs]: Created new folder Interfaces. All the interfaces are moved there.

File: E-Commerce-Server/ECom.Constants/ValidationConsts.cs (hunk 13 above)
Comment (ViktorKapra) [reply E-Commerce-Server/ECom.Constants/ValidationConsts.cs]: Constants were renamed.

**Review by hottab9595 — APPROVED** [review]

**Pull request merged**